js_modules/dagster-ui/packages/ui-core/src/automation/MergedAutomationRoot.tsx

```diff
@@ -27,11 +48,44 @@ export const MergedAutomationRoot = () => {
     data: cachedData,
   } = useContext(WorkspaceContext);
 
-  const [searchValue] = useQueryPersistedState<string>({
+  const [searchValue, setSearchValue] = useQueryPersistedState<string>({
     queryKey: 'search',
     defaults: {search: ''},
   });
 
+  const [automationTypes, setAutomationTypes] = useQueryPersistedState<Set<AutomationType>>({
+    encode: (vals) => ({automationType: vals.size ? Array.from(vals).join(',') : undefined}),
+    decode: (qs) => new Set((qs.automationType?.split(',') as AutomationType[]) || []),
+  });
```

Comment:
I think our users are starting to expect that all filters be encoded in URL state. Can we add the 2 filters to the URL too?

Reply:
It's already in place for instigation state, via `useInstigationStatusFilter`.

For code locations, I'm just using the existing `useCodeLocationFilter`, so maybe that's the right place to make that change if we want to make it more widespread after all...